packages/ui/src/components/Channel/index.tsx

```diff
@@ -270,124 +270,125 @@ export function Channel({
-                      // showAddButton={
-                      //   !isChatChannel && canWrite && !showBigInput
-                      // }
-                      // onPressAddButton={() =>
-                      //   channel.type === 'gallery'
-                      //     ? setShowAddGalleryPost(true)
-                      //     : setShowBigInput(true)
-                      // }
-                      goToSearch={goToSearch}
-                      showSpinner={isLoadingPosts}
-                      showMenuButton={true}
-                    />
-                    <YStack alignItems="stretch" flex={1}>
-                      <AnimatePresence>
-                        {draftInputPresentationMode !== 'fullscreen' && (
-                          <View flex={1}>
-                            {channel && posts && (
-                              <Scroller
-                                key={scrollerAnchor?.postId}
-                                inverted={isChatChannel ? true : false}
-                                renderItem={renderItem}
-                                renderEmptyComponent={renderEmptyComponent}
-                                anchor={scrollerAnchor}
-                                posts={posts}
-                                hasNewerPosts={hasNewerPosts}
-                                hasOlderPosts={hasOlderPosts}
-                                editingPost={editingPost}
-                                setEditingPost={setEditingPost}
-                                editPost={editPost}
-                                channelType={channel.type}
-                                channelId={channel.id}
-                                firstUnreadId={
-                                  initialChannelUnread?.countWithoutThreads ??
-                                  0 > 0
-                                    ? initialChannelUnread?.firstUnreadPostId
-                                    : null
-                                }
-                                unreadCount={
-                                  initialChannelUnread?.countWithoutThreads ?? 0
-                                }
-                                onPressPost={
-                                  isChatChannel ? undefined : goToPost
-                                }
-                                onPressReplies={goToPost}
-                                onPressImage={goToImageViewer}
-                                onEndReached={onScrollEndReached}
-                                onStartReached={onScrollStartReached}
-                                onPressRetry={onPressRetry}
-                                onPressDelete={onPressDelete}
-                                activeMessage={activeMessage}
-                                setActiveMessage={setActiveMessage}
-                                ref={flatListRef}
-                              />
+                    <ChannelHeaderItemsProvider>
```

Comment:
big hunk for small change:
- added `ChannelHeaderItemsProvider` (and a fragment `<>` as children to appease type check)
- removed the commented-out `showAddButton`/`onPressAddButton`